schema/v1-example-statement.json

```diff
@@ -0,0 +1,36 @@
+{
+  "_type": "https://in-toto.io/Statement/v0.1",
+  "subject": [
+    {
+      "name": "oak_functions_loader",
+      "digest": {
+        "sha256": "d4d5899a3868fbb6ae1856c3e55a32ce35913de3956d1973caccd37bd0174fa2"
+      }
+    }
+  ],
+  "predicateType": "https://slsa.dev/provenance/v0.2",
+  "predicate": {
+    "builder": {
+      "id": "https://github.com/project-oak/transparent-release"
+    },
+    "buildType": "https://github.com/project-oak/transparent-release/amber-slsa-buildtype/v1.json",
```

Comment:
I take it that the assumption here is that the docker image is mounted in a specific way, and that the git commit is cloned at a specific folder, and this command is executed in a specific manner and by a specific user? And in principle, if we change any of these things we would have to bump the version of the build type, since it would be incompatible with the previous version, right? Not to worry about that for now, but that's also what I meant when I said that there may probably be some SLSA provenance format that describes a generic way of putting together a docker image with a git commit and run a command (either now or in the future).

Reply:
In the future there likely will be some standard for that, yes. There isn't one yet, we're all still in the initial experimental phase.

Ideally all the actually relevant "specific ways, manners, etc" relevant for a build should be captured in this build type. But you're right that it cannot actually be truly comprehensive and defacto there are and likely always will be implicit defaults. We'll bump versions as needed with changes to those. Ideally that iterative process will eventually yield the the stable, "generic way" you alluded to.